src/utils/testing.js

```diff
@@ -22,17 +22,46 @@
 
 import MCT from 'MCT';
 
+const TIME_SYSTEM_OPTIONS = {
+    clock: 'fixed',
+    timeSystemKey: 'utc',
+    bounds: {
+        start: 0,
+        end: 1
+    }
+};
+
 let nativeFunctions = [];
 let mockObjects = setMockObjects();
+let openmct;
+let readyCallback;
+let timeSystemKey;
+let validate;
+let start;
+let end;
+
+export function waitForRouter(done, validateHashFunc) {
+    readyCallback = done;
+    validate = validateHashFunc;
+
+    return () => {
+        openmct.router.on('change:hash', checkHash);
+    };
+}
+
+export function createOpenMct(timeSystemOptions = TIME_SYSTEM_OPTIONS) {
```

Comment:
This has become quite complicated. I'm having trouble understanding how `waitForRouter` works, especially WRT to validation of the hash, which is at least partly because of the use of global variables. If developers can't understand it, they will make mistakes, and because we're using global variables so extensively here this will lead to mistakes and memory leaks.

Reply:
reverted back to old version